File: Treasury.Application/DTOs/BudgetSectionDto.cs
namespace Treasury.Application.DTOs;

public class BudgetSectionDto
{
    public int Id { get; set; }

    public string Name { get; set; }

    public int NumOfItems { get; set; }

    public decimal AmountRequested { get; set; }

    public decimal AmountProposed { get; set; }

    public bool Appealed { get; set; }

    public decimal RequestedAppeal { get; set; }

    public decimal ApprovedAppeal { get; set; }

    public decimal AmountApproved => AmountProposed + ApprovedAppeal;
}

File: src/main/java/edu/wpi/sga/treasury/application/dto/budget/BudgetSectionDto.java
package edu.wpi.sga.treasury.application.dto.budget;

import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;

@Getter
@Setter
@Builder
public class BudgetSectionDto {
    private Integer id;
    private String sectionName;
    private Integer numOfItems;
    private BigDecimal amountRequested;
    private BigDecimal amountProposed;
    private BigDecimal amountApproved;

    private boolean appealed;
    private BigDecimal appealAmount;
    private String appealDecision;
    private BigDecimal approvedAppeal;
}
